### DISCUSSION TITLE
Project wide diagnostics picker

### DISCUSSION THREAD

**Discussion by peteringram0** [helix-editor/helix#7254] [Q&A] [answered]
with space "mode + D" diagnostics are only showing for the file that is open. Is there any way to get diagnostics for the whole project?

**Comment by pascalkuthe** [comment, marked as answer]
space-D does show all doganostics aens by the language server. If your language server only sends diagnostics for open files then that's a limitation o f your LS and not helix. With rust-analyzer space-D will show all diagnostics within the workspace for example

**Reply by peteringram0** [reply]
Is this supported with typescript-language-server? I get the following health:
typescript               ✓ typescript-language…   None                     ✓                        ✓                        ✓

**Reply by brianmay** [reply]
For me space-D only shows the current file when using it with rust-analyzer.

**Reply by deifactor** [reply]
Lowercase d shows the current file, uppercase D shows all diagnostics.

**Comment by peteringram0** [comment]
@pascalkuthe Do you if there is any way to adjust the ts-lang-server to send project wide diagnostics at all? Having to load every file into buffer in a large project wont work for me.

**Reply by the-mikedavis** [reply]
It looks like it's an open issue upstream: https://github.com/typescript-language-server/typescript-language-server/issues/358

Sounds like it should be possible to support in the server, it just needs to be implemented.
